### DISCUSSION TITLE
Configurations in config file

### DISCUSSION THREAD

**Discussion by Gubbledenut** [bazmonk/digigurdy-baz#70] [Ideas]
I have been using additional key layout for my 31 key tenor now in a modified  file, so I only need to add it to the folder before compiling.
My config file also includes the remapping of button 6 
Next work in progress is adding the alternative key layouts for my 36 key digiharpa and 26 key digimora ,
They will be commented out as needed.
The fun part is establishing the buttons 1 to 6 , I think I’ll be asking questions later..in short on the Harpa the keys are set chromatically the three rows act on each of the three strings, top row highest string to bottom row to third string, 4th string is just a drone until I build the 4 row version! 
Tunings will also need to be adjusted and added too.

**Comment by bazmonk** [comment]
Sorry... I kinda took the weekend off from doing anything.

I saw you mentioned the 1-6 mapping on Facebook, too.  Can you show me a pic of the keybox itself, and the key array you're using, and which ones you want to be 1-6?

**Comment by Gubbledenut** [comment]
Sorted, I need to update my keymap drawings to include the key index as well as each key's note, key no and teensy pin. But here are the corrected arrays for the 3 layouts I currently have, there will be a 4th the moraharpa, one day 

// KEYBOX VARIABLES: DJ Added a few
//Uncomment for required Keyboard layout
// The pin_array[] index here represents the MIDI note offset, and the value is the corresponding
// teensy pin.  This defines which physical keys are part of the "keybox" and what order they're in.
//
// NOTE: There's no key that produces 0 offset (an "open" string),
// so the first element is bogus.  It gets skipped entirely.

/*
// 24 Key Standard Hurdy Gurdy
int pin_array[] = {-1, 2, 24, 3, 25, 26, 4, 27, 5, 28, 29, 6, 30,
                   7, 31, 8, 32, 33, 18, 34, 19, 35, 36, 20, 37};

// This is literally just the size of the above array minus one.  I need this as a const to
// declare the KeyboxButton array later on... or I just don't know enough C++ to know how to
// *not* need it ;-)
const int num_keys = 24;

// These control which buttons on the keybox have the roles of the X, O, 1-6, etc. buttons.
// Users with non-"standard" keyboxes (if there is such a thing!) may need to adjust these.
//
// These are array indexes, so if you count chromatically up your keybox and then subtract 1,
// that is its index.
const int BACK_INDEX = 0;
const int OK_INDEX = num_keys - 2;
const int BUTTON_1_INDEX = 1;
const int BUTTON_2_INDEX = 3;
const int BUTTON_3_INDEX = 4;
const int BUTTON_4_INDEX = 6;
const int BUTTON_5_INDEX = 8;
const int BUTTON_6_INDEX = 9;
const int TPOSE_UP_INDEX = num_keys - 1;
const int TPOSE_DN_INDEX = num_keys - 3;
const int CAPO_INDEX = num_keys - 5;
*/

/*
//Tenor 31 Key Hurdy Gurdy
 int pin_array[] = {-1, 54, 53, 52, 51, 50, 49, 48, 2, 24, 3, 25, 26, 4, 27, 5, 28, 29, 6, 30,
                   7, 31, 8, 32, 33, 18, 34, 19, 35, 36, 20, 37}; 
                   
const int num_keys = 31;

const int BACK_INDEX = 0;
const int OK_INDEX = num_keys - 2;
const int BUTTON_1_INDEX = 1;
const int BUTTON_2_INDEX = 3;
const int BUTTON_3_INDEX = 4;
const int BUTTON_4_INDEX = 6;
const int BUTTON_5_INDEX = 8;
const int BUTTON_6_INDEX = 10;
const int TPOSE_UP_INDEX = num_keys - 1;
const int TPOSE_DN_INDEX = num_keys - 3;
const int CAPO_INDEX = num_keys - 5; 
*/

//Nyckelharpa 3 Row 35 Key 
int pin_array[] = {-1, 2, 24, 3, 25, 26, 4, 27, //Bottom Row
                    5, 28, 29, 6, 30, 7, 31, 8, //Middle Row
                    32, 33, 41, 34, 42, 35, 36, 20, 37, 38, 0, 43,44,45,46,47,48,49,50,51}; //Top Row

const int num_keys = 35;

const int BACK_INDEX = 14;  //Top Row Left
const int OK_INDEX = 34;   // Top Row Right
const int BUTTON_1_INDEX = 0; //Bottom Row Left
const int BUTTON_2_INDEX = 1;
const int BUTTON_3_INDEX = 2;
const int BUTTON_4_INDEX = 3;
const int BUTTON_5_INDEX = 4;
const int BUTTON_6_INDEX = 5;
const int TPOSE_UP_INDEX = num_keys - 1;
const int TPOSE_DN_INDEX = num_keys - 3;
const int CAPO_INDEX = num_keys - 5;
